### PR TITLE
Fix vault bats tests for running against v0.0.6 of the provider

### PR DESCRIPTION
**Pull request by tomhjp**
**What this PR does / why we need it**:

v0.0.5 of the vault provider introduced a breaking change which the integration tests here were never updated for. This PR fixes the tests so they should pass for v0.0.6, and allow us to merge https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/pull/53 safely.

**Which issue(s) this PR fixes**:
None

**Special notes for your reviewer**:

Apologies for the conflicts with #245 but I've tried to keep the diff to a minimum to make it easier to port over to that PR.

### CODE REVIEW

**Comment by tomhjp** [comment]
CLA signed, although I may have to rewrite my commit email address

**Comment by aramase** [comment]
> v0.0.5 of the vault provider introduced a breaking change which the integration tests here were never updated for. This PR fixes the tests so they should pass for v0.0.6, and allow us to merge hashicorp/secrets-store-csi-driver-provider-vault#53 safely.

@tomhjp Is this breaking change and impact on existing `SecretProviderClass` documented?

File: test/bats/vault.bats [hunk 1]
```diff
@@ -86,17 +86,21 @@ EOF
   assert_success
 
   run kubectl exec -ti $VAULT_POD -- vault policy write example-readonly -<<EOF
+path "sys/mounts" {
```
Comment (aramase) [inline test/bats/vault.bats:89]: Is this a new policy required starting from `0.0.5` release of the vault provider?

**Review by aramase — COMMENTED** [review]
/lgtm

**Comment by tomhjp** [comment]
> Is this breaking change and impact on existing SecretProviderClass documented?

There's a note in the [readme](https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/#validate-the-secret), and also in the [release notes](https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/releases/tag/0.0.5) for 0.0.5.

File: test/bats/vault.bats (hunk 1 above)
Comment (tomhjp) [reply test/bats/vault.bats:89]: Sorry I missed this before. Yes, that's right, it does a read to check whether the `kv` engine is v1 or v2 starting from 0.0.5

**Pull request merged**